### PR TITLE
arm: final release notes for Cortex-R and Cortex-M

### PR DESCRIPTION
**Pull request by ioannisg**
v2.6.0 release notes for Cortex-R and final entries for Cortex-M.

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Can you remove the blank lines between bullets.

File: doc/releases/release-notes-2.6.rst [hunk 1]
```diff
@@ -181,10 +181,20 @@ Architectures
...
+    * Added initial support for Arm v8.1-m and Cortex-M55.
 
     * Added support for preempting threads while they are performing secure calls in Cortex-M.
 
+    * Added support for memory region generation by the linker based on device tree
+      node information in Cortex-M.
+
+    * Cleaned up definitions of SoC-specific memory regions in the common Cortex-M linker script.
+
+    * Added support for clearing NXP MPU region configuraton during Zephyr early boot stage.
```
Comment (bbolen) [inline doc/releases/release-notes-2.6.rst:193]: "configuration"

**Comment by ioannisg** [comment]
> Can you remove the blank lines between bullets.

Will remove also for the other ARCHes, then :)

File: doc/releases/release-notes-2.6.rst (hunk 1 above)
Comment (ioannisg) [reply doc/releases/release-notes-2.6.rst]: thanks.
and btw, if I missed any key changes for Cortex-R, let me know.

**Review by bbolen — APPROVED** [review]

**Pull request merged**